1.10/tutorials/dcos-101/service-discovery.md

```diff
@@ -11,18 +11,32 @@ menu_order: 4
 
 # Objective
-Your application in the previous part used `redis.marathon.l4lb.thisdcos.directory:6379` as the address for connecting to redis. As redis might be running on any agent in the cluster (and furthermore on different ports), how does this address link to the actual running redis instance?
-In this section, you will learn about DC/OS service discovery by exploring the different options for service discovery for apps in DC/OS.
+Your [app](https://raw.githubusercontent.com/joerg84/dcos-101/master/app1/app1.py) in the previous part of this tutorial used `redis.marathon.l4lb.thisdcos.directory` as the address for connecting to Redis, with a port of 6379. As Redis might be running on any agent in the cluster, and potentially on different ports, how does this address resolve to the actual running Redis instance?
 
-# Steps
-  [Service discovery](/docs/1.10/networking/) allows you to connect to your applications without necessarily knowing where they are running. Service discovery is particularly useful in cases where applications may fail and be restarted on a different host.
+In this section, you will learn about DC/OS service discovery by exploring the different options available for apps in DC/OS.
 
-  DC/OS provides two options for service discovery: Mesos-DNS and Named virtual IPs.
-  * SSH into your cluster to see how these service discovery methods work: `dcos node ssh --master-proxy --leader`,
-  <a name="mesosdns"></a>
-  * [Mesos-DNS](/docs/1.10/networking/mesos-dns/) assigns a Mesos-DNS for every Marathon app. The naming pattern is  *task.scheduler.mesos* and the default scheduler for jobs is `marathon`, so the Mesos-DNS name for your redis service is *redis.marathon.mesos*.
+# Service Discovery
+  [Service discovery](/docs/1.10/networking/) allows you to connect to your applications without necessarily knowing where they are running. This is particularly useful in cases where applications may fail and be restarted on a different host.
 
-  Let's use [dig](https://linux.die.net/man/1/dig) to retrieve the address record (also called the A record): `dig redis.marathon.mesos`.
+  DC/OS provides two options for service discovery: 
+
+  1. Mesos-DNS
+  1. Named Virtual IPs.
+
+
+SSH into the Mesos master node in your cluster to see how these different service discovery methods work: 
+
+`dcos node ssh --master-proxy --leader`
+
+# Mesos-DNS
+  
+  [Mesos-DNS](/docs/1.10/networking/mesos-dns/) assigns DNS entries for every Marathon app. The naming pattern for these entries is  *task.scheduler.mesos*
```

Comment:
assign DNS entries–which are resolvable from any node in the cluster–  

s/for every Marathon app/for every task.